A71-xds-fallback.md

```diff
@@ -127,94 +52,160 @@ to [xDS Protocol Specification][resource-does-not-exist].
+using the data from the fallback xDS control plane which is not desirable (see
+below).
 
-1. For each target, we need to store an ordered list of channels instead of
-just one channel.
-1. Whenever we lose contact with a given server and there is at least
-one resource requested but not cached or have previously lost contact and
-an uncached resource is requested, we create a channel/stream for the next
-server in the list unless we are at the last one.
-1. Whenever we get back in contact with a given server, if we have
-a channel/stream for a server later in the list, close it.
+Changes XdsClient to per-data plane target would enable gRPC to switch to
+fallback configuration only for channels that need xDS resource that have not
+been downloaded yet.
 
-Currently bootstrap JSON supports multiple xDS servers but semantics are not
-explicitly specified.
+This change would change the channels to get XdsClient for their data plane
+target. It will also require changes to the CSDS service to aggregate configs
+from multiple XdsClients. For gRPC Core, this would entail updating
+`grpc_dump_xds_configs` function to iterate over XdsClients and aggregate the
+data available in each. A new field will be added to the CSDS response
+to indicate the channel target the data is associated with. See [A40] for
+details on CSDS.
 
-1. xDS servers will be attempted in the order they are specified in
-the bootstrap JSON. Server will only be attempted if the previous entry in
-the list is not available.
-1. xDS client will report a failure if the last entry in the list is
-not available.
+gRPC servers using the xDS configuration will share the same XdsClient instance
+keyed with a dedicated well-known key value.
 
-In some gRPC implementations, the internal data structures do not allow for more
-than a single xDS server. The implementation needs to be updated to handle
-multiple servers, maintaining their fallback order.
+### Change xDS Bootstrap to handle multiple xDS configuration servers
 
-#### Add multi-client support to CSDS
+Current gRPC xDS implementations only use the first xDS configuration server
+listed in the bootstrap JSON document. Fallback implementation requires changes
+to use server listed and keep track of their ordering as it defines the
+relative priority of the different xDS config servers.
 
-See [A40] for details on CSDS.
+### Change XdsClient to support fallback within each xDS authority
 
-Add add a field to the ClientConfig message in the CSDS response to indicate
-which channel target the data is associated with.
+The fallback process is initiated if both of the following conditions hold:
 
-### Temporary environment variable protection
+* There is a connectivity failure on the ADS stream, as described in [A57]:
+    - Channel reports TRANSIENT_FAILURE.
+    - The ADS stream was closed before the first server response had been
+    received.
+* At least one watcher exists for a resource that is not cached.
```

Comment:
We still need to explicitly state what "cached" means in this context.  It looks like you mention that in the section on lines 43-53, but I think that content should be moved to this section instead.